File: VB/DXApplication191/SkinButtonHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.LookAndFeel
Imports DevExpress.Skins
Imports DevExpress.Utils
Imports DevExpress.Utils.Drawing

Namespace DXApplication191
    Friend Class SkinButtonPaintHelper
        Private _appearance As DevExpress.Utils.AppearanceObject
        Private _image As Image
        Private _buttonRectangle As Rectangle
        Public Property AppearanceObject() As AppearanceObject
            Get
                If _appearance Is Nothing Then
                    _appearance = New AppearanceObject()
                End If
                Return _appearance
            End Get
            Set(ByVal value As AppearanceObject)
                _appearance = value
            End Set
        End Property
        Public Property ButtonImage() As Image
            Get
                Return _image
            End Get
            Set(ByVal value As Image)
                _image = value
            End Set
        End Property
        Public Property ButtonRectangle() As Rectangle
            Get
                Return _buttonRectangle
            End Get
            Set(ByVal value As Rectangle)
                _buttonRectangle = value
            End Set
        End Property
        Public Property ButtonText() As String
        Public Sub New()
            AppearanceObject.TextOptions.HAlignment = DevExpress.Utils.HorzAlignment.Center
            AppearanceObject.TextOptions.VAlignment = DevExpress.Utils.VertAlignment.Center
        End Sub
        Public Sub DrawButton(ByVal args As PaintEventArgs, ByVal state As ObjectState)
            Using cache As New GraphicsCache(args)
                Dim element As SkinElement = CommonSkins.GetSkin(UserLookAndFeel.Default)(CommonSkins.SkinButton)
                Dim info As New SkinElementInfo(element, ButtonRectangle)

                If state = ObjectState.Normal Then
                    info.ImageIndex = 0
                End If
                If state = ObjectState.Hot Then
                    info.ImageIndex = 1
                End If
                If state = ObjectState.Pressed Then
                    info.ImageIndex = 2
                End If
                DrawSkinImage(cache, info)
                DrawText(cache)
                DrawImage(cache)
            End Using
        End Sub

        Private Sub DrawSkinImage(ByVal cache As GraphicsCache, ByVal info As SkinElementInfo)
            ObjectPainter.DrawObject(cache, SkinElementPainter.Default, info)
        End Sub

        Private Sub DrawText(ByVal cache As GraphicsCache)
            AppearanceObject.DrawString(cache, ButtonText, ButtonRectangle)
        End Sub
        Private Sub DrawImage(ByVal cache As GraphicsCache)
            If ButtonImage Is Nothing Then
                Return
            End If
            Dim imageBounds As Rectangle = GetImageBounds(ButtonRectangle, ButtonImage.Size)
            cache.Graphics.DrawImage(ButtonImage, imageBounds)
        End Sub

        Private Function GetImageBounds(ByVal buttonBounds As Rectangle, ByVal imageSize As Size) As Rectangle
            Return ImageLayoutHelper.GetImageBounds(buttonBounds, imageSize, ImageScaleMode.Clip, ImageAlignmentMode.MiddleLeft)
        End Function
    End Class
End Namespace

File: CS/DXApplication191/SkinButtonHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.LookAndFeel;
using DevExpress.Skins;
using DevExpress.Utils;
using DevExpress.Utils.Drawing;

namespace DXApplication191 {
    class SkinButtonPaintHelper {
        DevExpress.Utils.AppearanceObject _appearance;
        Image _image;
        Rectangle _buttonRectangle;
        public AppearanceObject AppearanceObject {
            get {
                if (_appearance == null)
                    _appearance = new AppearanceObject();
                return _appearance;
            }
            set {
                _appearance = value;
            }
        }
        public Image ButtonImage {
            get { return _image; }
            set { _image = value; }
        }
        public Rectangle ButtonRectangle { get { return _buttonRectangle; } set { _buttonRectangle = value; } }
        public string ButtonText { get; set; }
        public SkinButtonPaintHelper() {
            AppearanceObject.TextOptions.HAlignment = DevExpress.Utils.HorzAlignment.Center;
            AppearanceObject.TextOptions.VAlignment = DevExpress.Utils.VertAlignment.Center;
        }
        public void DrawButton(PaintEventArgs args, ObjectState state) {
            using (GraphicsCache cache = new GraphicsCache(args)) {
                SkinElement element = CommonSkins.GetSkin(UserLookAndFeel.Default)[CommonSkins.SkinButton];
                SkinElementInfo info = new SkinElementInfo(element, ButtonRectangle);

                if (state == ObjectState.Normal) info.ImageIndex = 0;
                if (state == ObjectState.Hot) info.ImageIndex = 1;
                if (state == ObjectState.Pressed) info.ImageIndex = 2;
                DrawSkinImage(cache, info);
                DrawText(cache);
                DrawImage(cache);
            }
        }

        void DrawSkinImage(GraphicsCache cache, SkinElementInfo info) {
            ObjectPainter.DrawObject(cache, SkinElementPainter.Default, info);
        }

        void DrawText(GraphicsCache cache) {
            AppearanceObject.DrawString(cache, ButtonText, ButtonRectangle);
        }
        void DrawImage(GraphicsCache cache) {
            if (ButtonImage == null) return;
            Rectangle imageBounds = GetImageBounds(ButtonRectangle, ButtonImage.Size);
            cache.Graphics.DrawImage(ButtonImage, imageBounds);
        }

        Rectangle GetImageBounds(Rectangle buttonBounds, Size imageSize) {
            return ImageLayoutHelper.GetImageBounds(buttonBounds, imageSize, ImageScaleMode.Clip, ImageAlignmentMode.MiddleLeft);
        }
    }
}
